### PR TITLE
eXist 5.3.0 compatible version

### PR DESCRIPTION
**Pull request by line-o**
crypto lib compatible with the coming version of eXist-db (5.3.0)

This version is built on top of the community project base but includes the latest changes and fixes made by @claudius108 

Thanks to @chakl for tying up the loose ends.

### CODE REVIEW

**Comment by joewiz** [comment]
Thank you, @chakl and @line-o!

Is there a reason the PR is against the eXist-5.3.0 branch instead of master?

Could you confirm what the minimum eXist compatible version is now?

Also, once this is merged and published, do you recommend that we pull the "v5.3.0" version from the public repo? I recall this was known to be incompatible with eXist 5.x, despite its dependency declarations. http://exist-db.org/exist/apps/public-repo/packages/expath-crypto-exist-lib.html?eXist-db-min-version=4.4.0

**Comment by line-o** [comment]
Version 5.3.0 of the crypto lib was, despite not being built from this repository, overwritten with a package that was version 1.0.0 with its version bumped. 
Pulling this package from the public repository was vetoed from the community.
The new version will therefore be 6.0.0 and will have a minimum processor dependency of 5.3.0

File: pom.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
```
Comment (adamretter) [inline pom.xml:11]: Did you mean to change the file perms on several of the files in the PR from 0644 -> 0755? I am not sure why you would want the execute bits set on non-executable files?

File: .classpath [hunk 2]
```diff
@@ -15,11 +15,13 @@
 		<attributes>
 			<attribute name="optional" value="true"/>
 			<attribute name="maven.pomderived" value="true"/>
+			<attribute name="test" value="true"/>
```
Comment (adamretter) [inline .classpath:18]: Eclipse Project files should not be in SCM. Can you remove them and also add appropriate rules to your `.gitignore` please?

**Review by adamretter — CHANGES_REQUESTED** [review]

File: .classpath (hunk 2 above)
Comment (line-o) [reply .classpath:18]: I do not really understand why this is worthy to block the merge of this PR.
As only one line was added (by @claudius108) it is clear that this file was in version control before.

File: pom.xml (hunk 1 above)
Comment (line-o) [reply pom.xml:11]: Yeah, I noticed those permission changes as well. But the goal of this commit is to make progress. 
As already discussed in a previous community call this package will be replaced by a version making use of the cryptographic capabilities bundled with exist in the foreseeable future.

Comment (line-o) [reply pom.xml:11]: Changing them would not be a big of a deal.

File: .classpath (hunk 2 above)
Comment (line-o) [reply .classpath:18]: in this commit 2a0cf88cde5cd412c48ea5ef1c3dd07a960d20e3

**Comment by line-o** [comment]
This PR needs to be closed as its target branch is master and not existdb 5.3.0

**Comment by line-o** [comment]
Superseded by #37

**Pull request closed without merging**